src/lib.rs

```diff
@@ -428,6 +428,22 @@ impl Build {
         self
     }
 
+    /// removes a flag from self.flags
```

Comment:
```suggestion
    /// Removes a flag added by [`Build::flag`].
```

Reply:
it removes flags directly from the Vec<Arc<&str>> of flags, so no, it doesn't just remove from Build::flag. it can also remove ar_flag, asm_flag, and (i think, haven't tried it) even file

Reply:
From [`cc::Build`](https://docs.rs/cc/latest/src/cc/lib.rs.html#95-135):

```rust
pub struct Build {
    flags: Vec<Arc<str>>,
    flags_supported: Vec<Arc<str>>,
    known_flag_support_status: Arc<Mutex<HashMap<String, bool>>>,
    ar_flags: Vec<Arc<str>>,
    asm_flags: Vec<Arc<str>>,
    no_default_flags: bool,
    files: Vec<Arc<Path>>,
    // ...
}
```

Since the function here only modifies `self.flags`, I don't think it will affect ar/asm flag or file

Reply:
@raisfeld-ori I still want this to change because the current doc includes implementation details and the function only removes flags added by `Build::flag`

Reply:
sorry, i didn't know that, i guess didn't read the code well enough, so i thought all flags would be part of a single structure. i'l change it now